src/Compilers/CSharp/Test/Semantic/Semantics/MethodTypeInferenceTests.cs

```diff
@@ -633,6 +635,155 @@ static void Main(string[] args)
                 Diagnostic(ErrorCode.ERR_BadArity, "Goo<>").WithArguments("Program.Goo<T, U>(T, U)", "method", "2"));
         }
 
+        [Fact]
+        public void Bug50782_2()
+        {
+            string source = @"
+#nullable enable
+
+interface IOperation<T> { }
+
+public class StringOperation : IOperation<string?> { }
+
+public class C {   
+    static void Main() {
+        TestF(out var _);
+        TestF(out _);
+    }
+   
+    static void TestF(out string result) => result = """";
```

Comment:
nit: consider using a raw string literal (using three double-quotes) for `source`. Then these quotes don't need to be escaped.